Component RatioLinkedAscertainment (#863)
CDCgov/PyRenew: docs/tutorials/ascertainment.qmd

#| label: setup
#| output: false

import jax.numpy as jnp
import numpyro.distributions as dist

from pyrenew.ascertainment import JointAscertainment, RatioLinkedAscertainment
from pyrenew.deterministic import DeterministicPMF, DeterministicVariable
from pyrenew.model import PyrenewBuilder
from pyrenew.observation import NegativeBinomialNoise, PopulationCounts
from pyrenew.randomvariable import DistributionalVariable
from pyrenew.time import MMWR_WEEK
```

## Overview

> An ascertainment rate is the probability that an infection results in an observed event.

Reviewer: ```suggestion
An ascertainment rate is the rate at which infections lead to observed events.
```

> For hospital admissions, this probability is often called the infection-hospitalization rate (IHR).

Reviewer: ```suggestion
For hospital admissions, this is often called the infection-hospitalization rate (IHR).
```

## Joint ascertainment

`JointAscertainment` places a multivariate normal prior on the logit-transformed ascertainment rates:

$$
  \boldsymbol{\eta}
  \sim
  \operatorname{MultivariateNormal}
  \left(
    \operatorname{logit}(\boldsymbol{\alpha}_0),
    \boldsymbol{\Sigma}
  \right),
  \qquad
  \boldsymbol{\alpha} = \operatorname{logit}^{-1}(\boldsymbol{\eta}).
$$

The baseline rates $\boldsymbol{\alpha}_0$ set the prior location.
The covariance matrix $\boldsymbol{\Sigma}$ controls the scale of variation in each logit-transformed rate and how strongly the rates vary together under the prior.

For two signals, positive prior covariance means that prior draws with a higher-than-expected rate for one signal tend to have a higher-than-expected rate for the other.
Negative covariance means that they tend to move in opposite directions.
Zero covariance means that the multivariate normal prior does not couple their deviations from the baseline rates.

The covariance structure is supplied to `JointAscertainment`; the component does not estimate an unknown covariance matrix from the observation data.

> It describes prior dependence, not a causal relationship between the signals.

Reviewer: One could infer $\Sigma$ rather than hard coding it, no? Just pass in a sampled quantity. I think simpler to say that `JointAscertainment` RVs treat $\Sigma$ as fixed; to infer $\Sigma$ you'd need to sample it upstream.

Author: good point. simplified.

A ratio below one means that hospitalization is less likely than an ED visit for a given infection; a ratio above one means that it is more likely.

```python
ratio_linked_ascertainment = RatioLinkedAscertainment(
    name="he_ascertainment",
    base_signal="ed_visits",
    linked_signal="hospital",
    base_rate_rv=DistributionalVariable("iedr", dist.Beta(1, 100)),
    ratio_rv=DistributionalVariable(
        "ihr_rel_iedr",
        dist.LogNormal(jnp.log(0.5), 0.35),
    ),
)
```

*Note*: These prior parameters are illustrative; they should be chosen using domain knowledge appropriate to the pathogen, population, observation definitions, and study period.

This model does not specify a covariance between the two signal rates directly.
However, the signal rates are not independent: the linked rate contains the sampled base rate.
The priors are assigned to the base rate and the ratio, and together they induce a distribution for the linked rate.

The ratio may be any positive value.
Unlike `JointAscertainment`, this component does not automatically constrain the derived linked rate to the unit interval.

> Users must choose priors that make $\alpha_{\mathrm{base}}\rho > 1$ negligibly likely.

Reviewer: You can imagine certain observables with more than one ascertained event per infection, so I don't think all ascertainment rates need to be probabilities. I think we can say something like

```suggestion
If both rates should be $<1$ (for instance because both are probabilities), make sure to choose priors that make $\alpha_{\mathrm{base}}\rho > 1$ negligibly likely.
```
